File: Go/74. Search a 2D Matrix.go
func searchMatrix(matrix [][]int, target int) bool {
    if (matrix == nil || len(matrix) == 0 || len(matrix[0]) == 0){
        return false;
    }
    row := len(matrix);
    rol := len(matrix[0]);
    l := 0;
    r := row * rol -1;
    for l < r {
        mid := l + ((r - l) >> 1);
        if matrix[mid / rol][mid % rol] >= target {
            r = mid;
        }else{
            l = mid + 1;
        }
    }
    if matrix[r / rol][r % rol] != target {
        return false;
    }
    return true;

}

File: src/Java/74. Search a 2D Matrix.java
class Solution {
    public boolean searchMatrix(int[][] matrix, int target) {
        if (matrix == null || matrix.length == 0 || matrix[0].length == 0){
            return false;
        }
        int raw = matrix.length;
        int col = matrix[0].length;
        int l = 0, r = col * raw -1;
        while(l < r){
            int mid = l + ((r-l) >> 1);
            if(matrix[mid / col][mid % col] >= target){
                r = mid;
            }else{
                l = mid + 1;
            }
        }
        if(matrix[l / col][l % col] != target){
            return false;
        }else{
            return true;
        }
    }
}